src/Common/src/TypeSystem/Common/TypeDesc.cs

```diff
@@ -601,5 +601,44 @@ public bool IsGenericDefinition
                 return HasInstantiation && IsTypeDefinition;
             }
         }
+
+        public uint GetSDIIndex()
```

Comment:
This should method should be a static (extension?) method that takes `TypeDesc` as argument. It should live somewhere near the JIT codegen, maybe start a new file under src/jitinterface/src.